File: yongkeun/6week/두개뽑아서더하기.js
function solution(numbers) {
  //다 더하고, 중복 제거하고, 정렬
  var answer = [];
  for (let i = 0; i < numbers.length; i++) {
    //모든 index를 돌음
    for (let j = 0; j < numbers.length; j++) {
      if (i !== j) {
        //혹시 i 와 j가 같다면 더하면 안됨, 0번 인덱스에 있는 2와 0번 인덱스에 있는 2를 더하는 것은 하지 않으므로 if문으로 같지 않을 때만 더하도록
        answer.push(numbers[i] + numbers[j]); //더한 값을 answer에 push
      }
    }
  }
  answer = [...new Set(answer)].sort((a, b) => a - b); //더한 모든 값이 answer에 있을텐데 그 값을 new Set을 이용해 중복을 제거해주고 sort를 이용해 오름차순으로 정렬함
  return answer;
}

File: dongmin/week6/두개 뽑아서 더하기.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] numbers) {      //반환형을 List형으로 바꾸면 List형으로 선언된 answer 배열로 변환하는 과정을 줄일 수 있다.
                                                        //다른 사람의 풀이를 보고 변환 과정 코드를 변경했더니 코드가 간결해졌다.

        Arrays.sort(numbers);
        List<Integer> result = new ArrayList<>();
        List<Integer> answer = new ArrayList<>();
        for (int i = 0; i < numbers.length-1; i++) {
            for (int j = i + 1; j < numbers.length; j++) {
                result.add(numbers[i]+numbers[j]);      //resut에 나올 수 있는 모든 값들을 담는다.
            }
        }
        Collections.sort(result);       //정렬을 통해 값들을 오름차순으로 정렬하고
        for (int num : result) {        
            if(!answer.contains(num))       //answer에 같은 값이 없을 때 
                answer.add(num);            //answer에 값을 넣는다.
        }

        return answer;
    }
}